### PR TITLE
LCAM-518

### PR DESCRIPTION
**Pull request by Mhowell494**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-518)

- Integrate crime-commons library
- Court Data API Schemas for contribution and correspondence state endpoints.
- New `MaatCourtDataService` providing methods for calling the Court Data API.

### CODE REVIEW

File: crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/MaatCourtDataServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package uk.gov.justice.laa.crime.contribution.service;
```
Comment (AafAnsari) [inline crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/MaatCourtDataServiceTest.java:1]: We could have some error/invalid params test cases here as these tests only covers the happy path.

**Review by AafAnsari — CHANGES_REQUESTED** [review]

File: crown-court-contribution/src/main/resources/schemas/CreateContributionRequest.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "applId": {
+      "type": "integer",
+      "description": "The associated application ID"
+    },
+    "userCreated": {
+      "type": "string",
+      "description": "The user creating the record"
+    }
+  },
```
Comment (loganathanMOJ) [inline crown-court-contribution/src/main/resources/schemas/CreateContributionRequest.json:20]: Missing laaTransactionId.

Comment (Mhowell494) [reply crown-court-contribution/src/main/resources/schemas/CreateContributionRequest.json:20]: This field is being passed as a header with all requests, which I think should be our preferred method. However, there's probably some work to do in the Court Data API to make sure we're consuming this and logging the value correctly.

File: crown-court-contribution/src/main/resources/schemas/ContributionRequest.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      "description": "Indicates if an income uplift was applied"
+    },
+    "basedOn": {
+      "type": "string",
+      "description": "Indicates what the contributions are based on (e.g. Means assessment)"
+    },
+    "transferStatus": {
+      "type": "string",
+      "enum": ["ERRATA", "QUARANTINE", "REQUESTED", "SENT", "MANUAL", "RECEIVED"],
+      "description": "The transfer status of the contribution record"
```
[the remark this answers is not in the document]
Comment (Mhowell494) [reply crown-court-contribution/src/main/resources/schemas/ContributionRequest.json:45]: Done!

File: crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/MaatCourtDataServiceTest.java (hunk 1 above)
Comment (Mhowell494) [reply crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/MaatCourtDataServiceTest.java:1]: I'm not convinced this is necessary here as the underlying REST client is thoroughly tested in the crime-commons library. I can't think what else we would want to test? AFAIK this is how the service classes are tested in all of our modern services.

**Review by loganathanMOJ — APPROVED** [review]

**Pull request merged**